tests/acceptance/test_build.py

```diff
@@ -483,12 +488,13 @@ def test_preferred_versions(self, prepared_test_build, recipe, version):
                         % (pn_style, base_recipe, version)
                     )
 
-            init_env_cmd = "cd %s && . oe-init-build-env %s" % (
-                prepared_test_build["bitbake_corebase"],
-                prepared_test_build["build_dir"],
-            )
             run_verbose("%s && bitbake %s" % (init_env_cmd, recipe))
 
+        # QA-784: setting/unsetting PREFERRED_VERSION and rebuilding the recipes breaks some times
+        # the state cache so that consecutive tests will fail in a very inexplicable way.
+        # Cleaning the cache here for each version built seems to workaround the issue...
+        run_verbose("%s && bitbake --cmd cleanall %s" % (init_env_cmd, recipe))
```

Comment:
Hmm, it's a pretty heavy handed fix, which keeps the build cache empty for our components for all build hosts. But I don't really have any better ideas either.

Another possible issue: I'm not sure how it is today, but last time I ran `cleansstate` (or `cleanall`, which calls it) with the mounted NFS cache we use, it took nearly an hour to complete. I believe this is primarily a function of the total cache size, therefore it will not improve with additional runs.

But this is at least a few years ago now, and is probably heavily dependent on locality of the NFS share. So it's mostly an FYI that I have seen this in the past, and you should maybe compare running time with a pristine pipeline just to be sure.